src/lex/mod.rs

```diff
@@ -118,14 +118,14 @@ impl Lexer {
         self.chars[self.location.offset as usize..].chars()
     }
 
-    fn slice(&self, span_start: u32) -> RcStr {
+    fn slice(&self, span_start: u32) -> ArcStr {
         use std::ops::Range;
-        RcStr::from(self.chars.clone())
-            .slice_with(|s| {
-                s.get::<Range<usize>>(self.span(span_start).span.into())
-                    .unwrap_or("")
-            })
-            .unwrap()
+
+        let chars = self.chars.clone();
+        let chars = chars
+            .get::<Range<usize>>(self.span(span_start).span.into())
+            .unwrap_or("");
+        ArcStr::from(chars)
```

Comment:
Doesn't this copy the entire token? The whole point of using Rc is so that we don't burn up memory by copying each token many times.

Reply:
This would copy the string. You want to clone `self.chars` since it's Rc then substring on that using ArcStr's substring function.

Reply:
Which `substring` function?

Reply:
Oh maybe the sub string was an open PR. You could wait for it to go through or what you can do is if you clone the Rc into an RcStr then do a `get` or something similar which gives you a `&str` and you can rebind that with the ArcStr or something like that. I can give you better details when I'm back at my computer.

Reply:
This is the PR- it's merged but not in a release yet. https://github.com/thomcc/arcstr/pull/12

Reply:
So this means you'll also want to use `Substr` type instead of a raw `ArcStr`.

Reply:
To unblock this, can you use the git repo for now? That way we can see if this at least works before having to wait for the release.

Reply:
Oh god. that will take time to refactor everything, because now almost everything has to use `Substr`

Reply:
Sorry. I think I've done this refactor for y'all though, it made me notice a couple missing bits in the API too.